server/routes/api/chart.py

```diff
@@ -147,61 +138,67 @@ def geojson(dcid):
-    in those chart configs
+def get_choropleth_configs():
+    """ Gets all the chart configs that have choropleth charts
 
     Returns:
-        a tuple consisting of:
-            set of all stat vars including both the stat vars of the charts and stat vars of denominators
-            list of chart configs that are choropleth chart configs
+        list of chart configs that are choropleth chart configs
     """
     chart_config = current_app.config['CHART_CONFIG']
-    all_sv = set()
     chart_configs = []
     for config in chart_config:
         if config.get('isChoropleth', False):
-            # we should only be making choropleths for configs with a single stat var
-            # TODO(chejennifer) add test for chart config to ensure isChoropleth is only added to charts with single statvar
-            sv = config['statsVars'][0]
-            all_sv.add(sv)
-            if 'relatedChart' in config and config['relatedChart'].get(
-                    'scale', False):
-                all_sv.add(config['relatedChart'].get('denominator',
-                                                      'Count_Person'))
-            if 'denominator' in config:
-                all_sv.add(config['denominator'][0])
             chart_configs.append(config)
-    return all_sv, chart_configs
+    return chart_configs
 
 
-def process_choropleth_data(all_sv_data):
-    """ For each combination of geo dcid and stat var, gets the source series containing
-    data from the most recent date
+def get_stat_vars(configs):
+    """  Gets all the stat vars and denominators in the given list of chart
+    configs
 
-    Args:
-        all_sv_data: value of the key 'placeData' in the object received from getStatsAll
+    Args: 
+        configs: list of chart configs
 
     Returns:
-        dictionary of dictionary of statvar: data for single source series, keyed by place dcid
+        tuple consisting of
+            set of all stat var dcids
+            set of all denominator stat var dcids
     """
-    result = {}
-    for geo, data_for_geo in all_sv_data.items():
-        result[geo] = dict()
-        for sv, sv_data in data_for_geo.get('statVarData', {}).items():
-            most_recent_date = None
-            for source in sv_data.get('sourceSeries', []):
-                source_values = source.get('val', {})
-                if len(source_values.keys()) < 1:
-                    continue
-                sorted_dates = sorted(source_values.keys())
-                if not most_recent_date or sorted_dates[-1] > most_recent_date:
-                    most_recent_date = sorted_dates[-1]
-                    result[geo][sv] = {
-                        'data': source_values,
-                        'provenanceUrl': source.get('provenanceUrl', '')
-                    }
-    return result
+    stat_vars = set()
+    denoms = set()
+    for config in configs:
+        stat_vars.add(config['statsVars'][0])
```

Comment:
check for existence of, and length of 'statsVars' first? or use .get so there is a default value to fall back on

Reply:
done